### PR TITLE
Issue770 sep arguments for reading csv files and storing reports

### LINKED ISSUE #770: sep argument for reading and storing files
GGIR users with different default settings to separate lists in their machines would benefit from a general `sep` argument to be used for reading/storing csv files.

### PR DESCRIPTION
**Pull request by jhmigueles**
Fixes #770 --> Includes sep_reports, sep_logs, and sep_config to read csv files (logs and config file) and to store reports.

Checklist before merging:

- [x] Existing tests still work (check by running the test suite, e.g. from RStudio).
- [ ] Added tests (if you added functionality) or fixed existing test (if you fixed a bug).
- [x] Updated or expanded the documentation.
- [x] Updated release notes in `inst/NEWS.Rd` with a user-readable summary. Please, include references to relevant issues or PR discussions.
- [ ] Added your name to the contributors lists in the `DESCRIPTION` and `CITATION.cff` files.

### CODE REVIEW

**Comment by vincentvanhees** [comment]
This branch is behind with the master branch, updating with master branch should address some or all of the failing tests

**Comment by jhmigueles** [comment]
Thanks Vincent, I was working on that. It was not only updating the branch but also changing some read.csv functions for data.table::fwrite (fwrite can handle the sep argument better than read.csv). Now the tests pass, but I have an extra proposal below.

The `data.table::fread` function has an automatic detection for `sep`. I have tested this using comma, semicolons, and tabs in the files and it works well, it can read the data correctly. Should we investigate if replacing all `read.csv` for `data.table::fread` in GGIR work properly? If so, we can skip the sep_logs and we might not need to include another `sep` argument to read in the csv files with raw accelerometer data in the future.

If you agree with this, I will replace all read.csv and write.csv for fread and fwrite, and write some extra tests to make sure GGIR can read correctly raw acc files and log files with different `sep`.

**Comment by vincentvanhees** [comment]
Ok, sounds good. Double check that output format is the same, as I believe fread produces data tables instead of data.frames as read.csv does. Either way, I suspect that that could be a potential source for problems.

**Comment by vincentvanhees** [comment]
No, don't change it there as that would mean I have to test the functionality again on all test files. So, this would have to be a separate issue and a seperate project.

**Comment by jhmigueles** [comment]
Great, so now it is working and all the tests pass. Only two new arguments are added: `sep_reports` and `sep_config`, and they are only used to store csv files. I am wondering if we could simplify this to only `sep` and it would be clarified in the documentation that it is used both for the reports and the config file. What do you think? 

As reading is now more flexible with `fread`, we do not need any other `sep` argument for reading logs.

File: R/g.dotorcomma.R [hunk 1]
```diff
@@ -26,7 +26,7 @@ g.dotorcomma = function(inputfile, dformat, mon, desiredtz = "", loadGENEActiv =
     # lot of zeros, which makes it impossible to detect decimal separator
     # "." will then be the default, which is not correct for "," systems.
     while (skiprows < 1000000) { #foundnonzero == FALSE & 
-      deci = as.matrix(read.csv(inputfile, skip = skiprows, nrow = 10))
+      deci = as.matrix(data.table::fread(inputfile, skip = skiprows, nrows = 10, data.table = FALSE))
       skiprows = skiprows + 10000
```
Comment (vincentvanhees) [inline R/g.dotorcomma.R:30]: Please undo this change to g.dotorcomma, as we discussed [here](https://github.com/wadpac/GGIR/pull/791#issuecomment-1566764332). It has taken a lot of time to develop and test the code for reading csv files. Changing it without addressing a clear bug and without a clear functional improvement seems like an unnecessary risk.

File: R/g.getstarttime.R [hunk 2]
```diff
@@ -47,7 +47,7 @@ g.getstarttime = function(datafile, P, header, mon, dformat, desiredtz, configtz
     starttime = as.POSIXlt(starttime)
   } else if (dformat == 2 & (mon == 3 | mon == 4 | mon == 6)) {
     if (mon == 3 | mon == 6) {
-      tmph = read.csv(datafile, nrow = 8, skip = 1)
+      tmph = data.table::fread(datafile, nrows = 8, skip = 1, data.table = FALSE)
       tmphi = 1
```
Comment (vincentvanhees) [inline R/g.getstarttime.R:51]: Please undo these change to g.getstarttime, as we discussed [here](https://github.com/wadpac/GGIR/pull/791#issuecomment-1566764332). It has taken a lot of time to develop and test the code for reading csv files. Changing it without addressing a clear bug and without a clear functional improvement seems like an unnecessary risk.

**Review by vincentvanhees — COMMENTED** [review]
Hi Jairo, please find my comments below. I think there are too many bug sensitive changes at the moment. I think it would be better to first do the `sep` update for reports, config file, and data cleaning file without touching the `sep` for reading accelerometer files. Also I would not deprecate the existing sep arguments yet. Once that works for a while, we can deprecate the existing sep arguments. And once that works we can some day consider changing the sep used for reading accelerometer csv files. I am worried that if we do all in one go there is a serious risk that many researchers will start running into problems and then it is us who have to clean it up.

File: R/g.part1.R [hunk 3]
```diff
@@ -274,7 +274,7 @@ g.part1 = function(datadir = c(), outputdir = c(), f0 = 1, f1 = c(),
       # three columns respectively named temperature.offset.x, temperature.offset.y, and temperature.offset.z
       # the end-user can generate this document based on calibration analysis done with the same accelerometer device.
       if (length(params_rawdata[["backup.cal.coef"]]) > 0 & check.backup.cal.coef == TRUE) {
-        bcc.data = read.csv(params_rawdata[["backup.cal.coef"]])
+        bcc.data = data.table::fread(params_rawdata[["backup.cal.coef"]], data.table = FALSE)
         if (isTRUE(params_rawdata[["do.cal"]]) & verbose == TRUE) {
```
[the remark this answers is not in the document]
Comment (jhmigueles) [reply R/g.part1.R:278]: I think this change is needed, it is used to read the csv with the calibration coefficients from the data_quality_report.csv. If this file has been saved with ";" or "\t" as sep, then it would be good to have a flexible function to read the data rather than read.csv.

**Comment by jhmigueles** [comment]
I have revised your proposed changes, see also a couple of comments above related to changes that I think it would be better to keep.

File: R/g.getstarttime.R [hunk 4]
```diff
@@ -101,7 +101,7 @@ g.getstarttime = function(datafile, P, header, mon, dformat, desiredtz, configtz
     options(digits.secs = 3)
     if ((mon == 3 & dformat != 6) | mon == 6) {
       options(warn = -1)
-      topline = as.matrix(colnames(as.matrix(read.csv(datafile, nrow = 1, skip = 0))))
+      topline = as.matrix(colnames(as.matrix(data.table::fread(datafile, nrows = 1, skip = 0, data.table = FALSE, check.names = TRUE))))
```
Comment (vincentvanhees) [inline R/g.getstarttime.R:104]: This change is also not necessary for now, correct?

File: R/load_params.R [hunk 5]
```diff
@@ -116,7 +116,8 @@ load_params = function(group = c("sleep", "metrics", "rawdata",
                           part5_agg2_60seconds = FALSE,
                           sensor.location = "wrist",
                           expand_tail_max_hours = NULL, recordingEndSleepHour = NULL,
-                          dataFormat = "raw", nonwear_approach = "2013")
+                          dataFormat = "raw", nonwear_approach = "2013",
+                          sep_reports = ",", sep_config = ",")
```
Comment (vincentvanhees) [inline R/load_params.R:120]: Should we move `sep_reports` from `params_general` to `params_output` as it is specific to the GGIR output?

Also I see that there are a couple more arguments in `params_general` that could move to other categories:
- `sep_config` is also a type of output, by which it could fit in `params_output` even though it is later used as input.
- `nonwear_approach` may fit better with `params_cleaning`.

I think we should try to avoid having too many arguments in the general category as that category is more intended to have all arguments that do not fit with any of the other categories.

**Review by vincentvanhees — CHANGES_REQUESTED** [review]
Thanks Jairo, two more minor comments.

**Review by vincentvanhees — CHANGES_REQUESTED** [review]
Thanks Jairo, two more minor comments.

**Comment by jhmigueles** [comment]
Already implemented these changes and just tested it again with sleeplogs in different formats and it wors as expected.

**Comment by vincentvanhees** [comment]
I have now updated this PR with the master branch, and made a few minor edits to this PR:
- Object `summary` was declared but not used, so I remove that line.
- I have renamed object `daysummary` to be P2daysummary_tmp. In this function this is important to clarify from which GGIR part a `summary` is as it loads and merges results from all the parts.

**Review by l-k- — APPROVED** [review]
This looks good to me, I didn't notice any issues during my testing or reading the code. I just sent a pull request with a small change to `test_gloadlog.R` that confirms that sleep logs with various separators can now be processed.

**Review by vincentvanhees — APPROVED** [review]
